channelserver/src/main.rs

```diff
@@ -61,27 +62,28 @@ fn channel_route(req: &HttpRequest<session::WsChannelSessionState>) -> Result<Ht
         Some(id) => {
             // if the id is valid, but not present, treat it like a "None"
             if id.len() == 0 {
-                Uuid::new_v4()
+                channelid::ChannelID::default()
             } else {
                 initial_connect = false;
-                Uuid::parse_str(id).unwrap_or_else(|e| {
+                let channel_id = channelid::ChannelID::from(id);
```

Comment:
Rather than having to remember to check `is_valid()`, would something like a `ChannelID::from_string(&str) -> Result<ChannelID, Error>` be useful here?

If not, maybe this logging can be merged with the second `is_valid()` call below rather than having to call it twice?

Reply:
Agreed on a from_string -> Result: force the error handling there vs requiring is_valid